public/content/mu-plugins/wp-admin-redirect.php

```diff
@@ -0,0 +1,15 @@
+<?php
+/*
+Plugin Name: WP-Admin Redirect
+Plugin URI:  https://github.com/felixarntz/wordpressdev
+Description: Redirects the plain /wp-admin/ URL to its actual location in the WordPress core subdirectory.
+Version:     1.0.0
+Author:      WordPressDev contributors
+Author URI:  https://github.com/felixarntz/wordpressdev
+*/
+
+if ( ! empty( $_SERVER['REQUEST_URI'] ) && 0 === strpos( $_SERVER['REQUEST_URI'], '/wp-admin' ) && admin_url() !== home_url( 'wp-admin/' ) ) {
+	// `wp_safe_redirect()` and `wp_redirect()` are not available here yet.
+	header( 'Location: ' . site_url( $_SERVER['REQUEST_URI'] ), true, 301 );
```

Comment:
Why not do this later at say `template_redirect` when `is_404()`?

Reply:
A bit slower, but able to re-use core functions. With the permanent redirect the browser should remember the redirect subsequently.

Not a big deal.

Reply:
Okay, I'll change this to be executed later and check `is_404()`. Then we also cover the weird case where someone might create a page with the slug `wp-admin` :)